packages/@uppy/aws-s3-multipart/src/MultipartUploader.js

```diff
@@ -219,32 +238,46 @@ class MultipartUploader {
     })
   }
 
-  _uploadPartRetryable (index) {
+  _prepareUploadParts (candidates) {
+    this.lockedCandidatesForBatch.push(...candidates)
```

Comment:
```suggestion
    this.lockedCandidatesForBatch.concat(candidates)
```

Reply:
would have to be `this.lockedCandidatesForBatch = this.lockedCandidatesForBatch.concat(candidates)` in that case :)

Reply:
IIRC `push(...candidates)` tends to perform better than `concat`+re-assignment.

Reply:
Yes we can ignore this suggestion